kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/BankParametersAccessServiceImpl.java

```diff
@@ -0,0 +1,759 @@
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isExcludedCustomerRefNbr(String)
+     */
+    public boolean isExcludedCustomerRefNbr(String custRefNumb) {
+        return getExcludedCustomerRefNbr().contains(custRefNumb);
+    }
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isExcludedByBaiAndAccount(int, String)
+     */
+    public boolean isExcludedByBaiAndAccount(int bai, String account) {
+        return getExcludedByBaiAndAccount().contains(bai + ":" + account);
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isSpecialBai(int)
+     */
+    public boolean isSpecialBai(int bai) {
+        return getSpecialBaiTypes().contains(bai);
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#isAllowedBai(int)
+     */
+    public boolean isAllowedBai(int bai) {
+        return getAllowedBaiTypes().contains(bai);
+    }
+
+
+    /**
+     * @see edu.arizona.kfs.fp.batch.service.BankParametersAccessService#getChartBankObjectCodeForDescriptionSearch(String)
+     */
+    public ChartBankObjectCode getChartBankObjectCodeForDescriptionSearch(String description) {
+
+        for (String descKey : getDescriptionSearchChartBankObjectCodeMap().keySet()) {
+            if (description.contains(descKey)) {
+                return descriptionSearchChartBankObjectCodeMap.get(descKey);
+            }
+        }
+        //not found, returning null;
+        return null;
+    }
+
+    /*
+     * Internal method that populates a Map with description as key and ChartBankObjectCode as values from the ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION parameter
+     */
+    protected Map<String, ChartBankObjectCode> getDescriptionSearchChartBankObjectCodeMap() {
+        if (descriptionSearchChartBankObjectCodeMap == null) {
+            descriptionSearchChartBankObjectCodeMap = new HashMap<String, ChartBankObjectCode>();
+            String acountObjByDescrParm = parameterService.getParameterValueAsString(DocumentCreationStep.class, KFSConstants.BankTransactionsParameters.ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION);
+            String[] acountObjByDescrList = acountObjByDescrParm.split(KFSConstants.BankTransactionsParameters.BANK_PARAMETER_DELIM);
+            if (acountObjByDescrList == null || acountObjByDescrList.length == 0) {
+                LOG.error("Parameter [ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION] has an empty value.  The process cannot be continued without this value.");
+                throw new RuntimeException("Parameter [ACCOUNT_AND_OBJECT_CODE_BY_DESCRIPTION] has an empty value.  The process cannot be continued without this value.");
+            }
+
+            int index = 0;
+            for (String chartBankObjectCodeMapStr : acountObjByDescrList) {
+                String[] tokens = chartBankObjectCodeMapStr.split("=");
+                if (tokens == null || tokens.length != 2) {
```

Comment:
Likewise a redundant check here (i.e., either line 493 blows up, the split happens and gives you multiple elements, or you end up with a 1 element array).